dune_api_scripts/duneanalytics.py

```diff
@@ -207,8 +193,19 @@ def query_result(self, result_id):
         self.session.headers.update({'authorization': f'Bearer {self.token}'})
 
         response = self.session.post(GRAPH_URL, json=query_data)
-        if response.status_code == 200:
-            data = response.json()
-            return data
-        print("Unsuccessful response", response.text)
-        return {}
+        return handle_dune_response(response)
+
+
+def handle_dune_response(response: Response, message: str = "successful response:"):
+    """
+    Parses response for errors by key and raises runtime error if they exist.
+    Successful responses will be printed to std-out and response json returned
+    :param response: Response from dune api post
+    :param message: optional custom message to be displayed upon success
+    :return: response in json format
+    """
+    response_json = response.json()
+    if 'errors' in response_json:
+        raise RuntimeError("Dune API Request failed with", response_json)
```

Comment:
To make debugging even easier, maybe we can even put the http_request into the message?

        raise RuntimeError("Dune API Request", query, " failed with", response_json)